addon/intl-computed-property.js

```diff
@@ -0,0 +1,34 @@
+import { get, defineProperty } from '@ember/object';
+import ComputedProperty from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+
+export default class IntlComputedProperty extends ComputedProperty {
+  constructor(fn, ...dependentKeys) {
+    super(function(propertyKey) {
+      let intl = get(this, 'intl');
+
+      assert(
+        `ember-intl: Could not look up 'intl' service for the property '${propertyKey}' on '${this}'.`,
+        intl
+      );
+
+      return fn.call(this, intl, propertyKey, this);
```

Comment:
Passing `this` as the third arguments allows derived classes to use arrow functions and still have access to the object instance the computed property is installed on.